### ISSUE TITLE
Handling of RAW images needs work

### ISSUE THREAD

**Issue by RhetTbull** [RhetTbull/osxphotos#218]
Not sure this is working correctly:

https://github.com/RhetTbull/osxphotos/blob/aa2cf826c7864a3872b68579a6bfef4566c47fe0/osxphotos/photosdb/photosdb.py#L2136-L2159

**Comment by RhetTbull** [comment]
```python
>>> photo = photosdb.get_photo("CE6BA7CA-D9AF-459C-A94A-7DD9669CF918")
>>> photo.path
'/Users/rhet/Pictures/Photos Library.photoslibrary/originals/C/CE6BA7CA-D9AF-459C-A94A-7DD9669CF918.cr2'
>>> photo.uti
'com.canon.cr2-raw-image'
>>> photo.path_raw
>>> photo._info["has_raw"]
False
>>> photo._info["UTI_raw"]
>>> ```

**Comment by RhetTbull** [comment]
See #183 ...apparently never tested these.  Doh!

**Comment by RhetTbull** [comment]
Seems to be working *if* RAW + JPEG imported.  If RAW only, `has_raw` is False.  Need a way to determine `is_raw`

**Comment by RhetTbull** [comment]
PhotoInfo.path_raw only returns correct path if a raw/jpeg pair.  It should always return path to the raw image even if there isn't a JPEG pair.

**Comment by RhetTbull** [comment]
Another example:
```python
>>> photosdb.db_path
'/Users/rhet/Pictures/Test-10.15.6.photoslibrary/database/photos.db'
>>> photo = photosdb.get_photo("D05A5FE3-15FB-49A1-A15D-AB3DA6F8B068")
>>> photo.uti
'com.adobe.raw-image'
>>> photo.path
'/Users/rhet/Pictures/Test-10.15.6.photoslibrary/originals/D/D05A5FE3-15FB-49A1-A15D-AB3DA6F8B068.dng'
>>> photo.path_raw
>>> photo.has_raw
False
>>>
```

**Comment by RhetTbull** [comment]
`has_raw` only applies to images that have JPEG/RAW complement.  Need to add `is_raw` to determine if image is actually RAW (w/o a JPEG complement).  I've not been able to find a way to determine this from the database other than looking at UTI.  It does appear that UTI for every RAW format I've tested includes "raw-image" so looking at that is a possibility:

```python
>>> for p in photos:
...     print(p.uti)
com.nikon.raw-image
com.panasonic.raw-image
com.adobe.raw-image
com.adobe.raw-image
com.kodak.raw-image
com.nikon.raw-image
com.adobe.raw-image
com.olympus.raw-image
com.konicaminolta.raw-image
com.pentax.raw-image
com.hasselblad.3fr-raw-image
com.adobe.raw-image
com.sony.arw-raw-image
com.epson.raw-image
com.adobe.raw-image
com.konicaminolta.raw-image
com.samsung.raw-image
com.canon.cr2-raw-image
com.canon.cr2-raw-image
com.sony.arw-raw-image
com.konicaminolta.raw-image
com.leafamerica.raw-image
com.nikon.raw-image
com.panasonic.rw2-raw-image
com.fuji.raw-image
com.sony.arw-raw-image
>>>
```

Lots of samples [here](https://www.rawsamples.ch/index.php/en/)

**Comment by neilpa** [comment]
There's also https://raw.pixls.us/ which is the successor to https://rawsamples.ch

**Comment by RhetTbull** [comment]
Fixed in convert_to_jpeg branch, v0.35.0
